pkcs7.go

```diff
@@ -147,6 +147,17 @@ func getOIDForEncryptionAlgorithm(pkey crypto.PrivateKey, OIDDigestAlg asn1.Obje
 	case *dsa.PrivateKey:
 		return OIDDigestAlgorithmDSA, nil
 	}
+	// XXX(fuhry@2022-10-20): when using a yubikey, smallstep/crypto's CreateDecrypter() returns a
+	// `*piv.keyRSA` from github.com/go-piv/piv-go/piv. Being a private type we can't handle this
+	// case directly, so this is my lazy way of working around it instead of forking piv-go or
+	// waiting for them to merge a PR.
+	//
+	// While this is a `crypto.Decrypter`, unfortunately that doesn't tell us anything about the
+	// algorithm (other than that it's not (EC)DSA, because *DSA doesn't support decryption)
+	switch fmt.Sprintf("%T", pkey) {
+	case "*piv.keyRSA":
+		return OIDEncryptionAlgorithmRSA, nil
+	}
```

Comment:
Instead of doing this, I would make it a little more generic, casting the pkey to a crypto.Signer and then calling the same method with an empty private key:

```go
switch k := pkey.(type) {
// ...
case crypto.Signer:
	switch k.Public() {
	case *rsa.PublicKey:
		return getOIDForEncryptionAlgorithm(&rsa.PrivateKey{}, OIDDigestAlg)
	case *ecdsa.PublicKey:
		return getOIDForEncryptionAlgorithm(&ecdsa.PrivateKey{}, OIDDigestAlg)
	case *dsa.PublicKey:
		return getOIDForEncryptionAlgorithm(&dsa.PrivateKey{}, OIDDigestAlg)
	}
}
```

In fact, I would try to get this merged this PR in https://github.com/mozilla-services/pkcs7, as at this moment I think both forks are in sync, and we can switch to it.